build/cmake/DefineOptions.cmake

```diff
@@ -62,8 +62,10 @@ endif()
 find_package(OpenSSL QUIET)
 CMAKE_DEPENDENT_OPTION(WITH_OPENSSL "Build with OpenSSL support" ON
                        "OPENSSL_FOUND" OFF)
-option(WITH_BOOSTTHREADS "Build with Boost thread support" OFF)
 option(WITH_STDTHREADS "Build with C++ std::thread support" OFF)
+CMAKE_DEPENDENT_OPTION(WITH_BOOSTTHREADS "Build with Boost threads support" ON
```

Comment:
Looks like a good idea for Windows.

For *nix, pthread has been the default, and IMO it should remain so because:
* *configure.ac* option states boost thread is experimental.
* In my experience, Thrift pthread implementation has better performance (although negligible for most cases) than that of boost or std-thread on Linux.